agent/consul/leader_routine_manager.go

```diff
@@ -17,13 +17,14 @@ type leaderRoutine struct {
 }
 
 type LeaderRoutineManager struct {
-	lock   sync.RWMutex
-	logger *log.Logger
+	lock    sync.RWMutex
+	logger  *log.Logger
+	logger2 hclog.Logger
 
 	routines map[string]*leaderRoutine
 }
 
-func NewLeaderRoutineManager(logger *log.Logger) *LeaderRoutineManager {
+func NewLeaderRoutineManager(logger *log.Logger, logger2 hclog.Logger) *LeaderRoutineManager {
 	if logger == nil {
```

Comment:
```suggestion
	if logger == nil || logger2 ==nil {
```

Reply:
Considered whether it was correct to use an `OR` expression here, decided that in all likelyhood the tuple `<logger, logger2>` should be either `<nil, nil>` or not, and a state like `<nil, initialized>` would not occur in practice.

Plus, the end goal is to remove the logger tuple anyways.